cpp_code/include/CrossCat/RandomNumberGenerator.h

```diff
@@ -20,26 +20,24 @@
 #ifndef GUARD_randomnumbergenerator_h
 #define GUARD_randomnumbergenerator_h
 
-#include <boost/random/mersenne_twister.hpp>
-#include <boost/random/uniform_01.hpp>
 #include <ctime>
-//
+#include <stdint.h>
+
 #include "constants.h"
+#include "weakprng.h"
 
-/////////////////////////
-// from runModel_v2.cpp
 class RandomNumberGenerator {
 public:
-    RandomNumberGenerator() : _engine(0), _dist(_engine) {};
-    RandomNumberGenerator(int SEED) : _engine(SEED), _dist(_engine) {};
+    RandomNumberGenerator(int seed = 0) { set_seed(seed); }
     double next();
-    int nexti(int max = MAX_INT);
+    int nexti(int bound = MAX_INT);
+    double stdnormal();
+    double stdgamma(double alpha);
```

Comment:
I started writing some statistical tests earlier this week (as noted in another review comment), but I'm not yet confident in their ability to detect plausible deviations from the intended distributions -- for example, detecting the modulo bias in the uniform integer distribution, or distinguishing a Student's t distribution from a normal distribution.  I'll post an update on that when it's ready.